fix: make timezone-dependent tests resilient to timezone boundaries

wakeensito opened the pull request:
- Fix dailyStatistics test in AbstractExecutionRepositoryTest to use fixed UTC dates instead of ZonedDateTime.now()
- Fix executionsCount test to use fixed UTC dates
- Fix metric aggregation tests in AbstractMetricRepositoryTest to use fixed UTC dates
- Fix time window test in AbstractMultipleConditionStorageTest to use resilient date assertions

This resolves the issue where tests fail when run during midnight in Europe/Prague timezone due to timezone boundary issues causing test data to be grouped into different days than expected.

The production code continues to use ZoneId.systemDefault() for consistent user experience.

wakeensito committed:
fix: make timezone-dependent tests resilient to timezone boundaries

- Fix dailyStatistics test in AbstractExecutionRepositoryTest to use fixed UTC dates instead of ZonedDateTime.now()
- Fix executionsCount test to use fixed UTC dates
- Fix metric aggregation tests in AbstractMetricRepositoryTest to use fixed UTC dates
- Fix time window test in AbstractMultipleConditionStorageTest to use resilient date assertions

This resolves the issue where tests fail when run during midnight in Europe/Prague timezone
due to timezone boundary issues causing test data to be grouped into different days than expected.

The production code continues to use ZoneId.systemDefault() for consistent user experience.

MilosPaunovic commented:
Hey @wakeensito,

Is this PR related to https://github.com/kestra-io/kestra/pull/11813? If so, does the other one need closing?

loicmathieu reviewed (approved):
LGTM, thanks

loicmathieu commented:
@wakeensito can you make sure it compiles?

wakeensito commented:
Hey @MilosPaunovic yes this PR is related to #11813, and @loicmathieu  I'll work on compile.

MilosPaunovic commented:
And do they both need to stay open, are they different?

wakeensito commented:
@MilosPaunovic no they both don't need to stay open

MilosPaunovic commented:
Okay then @wakeensito, can you close one of them?

wakeensito committed:
fix: use Collections.emptyList() instead of null for dailyStatistics state parameter

loicmathieu reviewed (approved):
LGTM

loicmathieu commented:
@wakeensito the test again didn't compile (see AbstractMetricRepositoryTest), can you make sure the modified tests compile and pass before pushing your branch to avoid wasting CI resources?

wakeensito committed:
Fix timezone bug in daily statistics & compile

- Use query timezone instead of system default for date formatting
- Fix AbstractJdbcExecutionRepository.fillDate() method
- Fix AbstractJdbcMetricRepository.fillDate() method
- Add dailyStatisticsEuropePragueMidnight() test case
- Fix missing ZoneId import in AbstractMetricRepositoryTest

Resolves issue where daily statistics were missing executions
that crossed midnight in non-server timezones (e.g., Europe/Prague).
